receiver/podmanreceiver/receiver.go

```diff
@@ -57,11 +63,51 @@ func newReceiver(
 		set:           set,
 	}
 
-	scrp, err := scraperhelper.NewScraper(typeStr, recv.scrape, scraperhelper.WithStart(recv.start))
+	return recv, err
+}
+
+func (r *receiver) RegisterMetricsConsumer(mc consumer.Metrics, set component.ReceiverCreateSettings) error {
+	r.metricsConsumer = mc
+	scrp, err := scraperhelper.NewScraper(typeStr, r.scrape, scraperhelper.WithStart(r.start))
 	if err != nil {
-		return nil, err
+		return err
+	}
+	r.metricsComponent, err = scraperhelper.NewScraperControllerReceiver(&r.config.ScraperControllerSettings, set, mc, scraperhelper.AddScraper(scrp))
+	return nil
+}
+
+func (r *receiver) RegisterLogsConsumer(lc consumer.Logs) {
+	r.logsConsumer = lc
+}
+
+
+func (r *receiver) Start(ctx context.Context, host component.Host) error {
```

Comment:
I think this methd should log warnings if either log or metrics consumer is not set up so users know they should do it.